packages/esm-patient-banner-app/src/banner-tags/print-identifier-sticker.component.tsx

```diff
@@ -0,0 +1,198 @@
+import React, { useCallback, useEffect, useMemo, useRef, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { Button, ModalBody, ModalFooter, ModalHeader } from '@carbon/react';
+import styles from './print-identifier-sticker.scss';
+import { useReactToPrint } from 'react-to-print';
+import { Column } from '@carbon/react';
+import { Grid } from '@carbon/react';
+import { age, displayName, showSnackbar, useConfig } from '@openmrs/esm-framework';
+
+interface PrintIdentifierStickerProps {
+  patient: fhir.Patient;
+  closeModal: () => void;
+}
+
+const PrintIdentifierSticker: React.FC<PrintIdentifierStickerProps> = ({ patient, closeModal }) => {
+  const { t } = useTranslation();
+  const [isPrinting, setIsPrinting] = useState(false);
+  const contentToPrintRef = useRef(null);
+  const [printRetryCount, setprintRetryCount] = useState(0);
+  const { printIdentifierStickerFields, printIdentifierStickerSize, excludePatientIdentifierCodeTypes } = useConfig();
+
+  const headerTitle = t('patientIdentifierSticker', 'Patient Identifier Sticker');
+
+  const onBeforeGetContentResolve = useRef(null);
+
+  useEffect(() => {
+    if (isPrinting && onBeforeGetContentResolve.current) {
+      onBeforeGetContentResolve.current();
+    }
+  }, [isPrinting]);
+
+  const patientDetails = useMemo(() => {
+    const getGender = (gender: string): string => {
+      switch (gender) {
+        case 'male':
+          return t('male', 'Male');
+        case 'female':
+          return t('female', 'Female');
+        case 'other':
+          return t('other', 'Other');
+        case 'unknown':
+          return t('unknown', 'Unknown');
+        default:
+          return gender;
+      }
+    };
+
+    const identifiers =
+      patient?.identifier?.filter(
+        (identifier) => !excludePatientIdentifierCodeTypes?.uuids.includes(identifier.type.coding[0].code),
+      ) ?? [];
+
+    return {
+      id: patient?.id,
+      photo: patient?.photo,
+      name: patient ? displayName(patient) : '',
+      dateOfBirth: patient.birthDate,
+      age: age(patient?.birthDate),
+      gender: getGender(patient?.gender),
+      address: patient?.address,
+      identifiers: identifiers?.length ? identifiers.map(({ value, type }) => ({ value, type })) : [],
+    };
+  }, [patient, t, excludePatientIdentifierCodeTypes?.uuids]);
+  const handleBeforeGetContent = useCallback(() => {
+    return new Promise<void>((resolve) => {
+      if (patient && headerTitle) {
+        onBeforeGetContentResolve.current = resolve;
+        setIsPrinting(true);
+        const printStyles = `@media print { @page { size: ${printIdentifierStickerSize}; } }`;
+
+        const style = document.createElement('style');
+        style.appendChild(document.createTextNode(printStyles));
```

Comment:
This seems messy. Let's move this into actual CSS and use a CSS var to set the size, that way we have something that's useful in more context than this. It would also avoid the issue where every time this callback is called, it will create yet another style node.

Reply:
Also, sequence-wise, I think DOM modifications like this need to be made in a `useEffect()` hook. As [the docs say](https://react.dev/reference/react/useEffect): "useEffect is a React Hook that lets you [synchronize a component with an external system.](https://react.dev/learn/synchronizing-with-effects)". In React's model, the DOM is always an "external system".

Reply:
Basically  `handleBeforeGetContent()` should probably just call `setIsPrintingTrue()` and a `useEffect()` hook should load the CSS, I guess (really it should just set the CSS variable).